tools/build/setup_macos_android.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#
+# Copyright 2022 The Outline Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+# Android SDK Build Tools:
+#   https://developer.android.com/studio/releases/build-tools.html
+# To find the latest version's label:
+#   sdkmanager --list|grep build-tools
+ANDROID_BUILD_TOOLS_VERSION=${ANDROID_BUILD_TOOLS_VERSION:-"30.0.3"}
+
+# NDK (side by side) version must be kept in sync with the default build tools NDK version.
+NDK_VERSION=${NDK_VERSION:-"21.0.6113669"}
+
+function install_jdk() {
+  if [[ -d "$HOME/Library/Java/JavaVirtualMachines/jdk-19.0.1.jdk" ]]; then
+    echo "JDK already installed"
+    return
+  fi
+
+  declare arch_url
+  if [[ "$(uname -m)" == "arm64" ]]; then
+    arch_url="aarch64"
+  else
+    acrh_url="x64"
+  fi
+  echo "Downloading JDK"
+  curl "https://download.java.net/java/GA/jdk19.0.1/afdd2e245b014143b62ccb916125e3ce/10/GPL/openjdk-19.0.1_macos-${arch_url}_bin.tar.gz" | tar -xzk -C "$HOME/Library/Java/JavaVirtualMachines/"
```

Comment:
can we prefer the Azul distribution? or allow specifying which, or maybe skipping the install_jdk step if $JAVA_HOME is already defined?

Reply:
We can discuss this, but it's out of scope of this PR.